waspc/data/Generator/templates/sdk/wasp/server/config.ts

```diff
@@ -28,6 +28,7 @@ type CommonConfig = BaseConfig & {
 
 type EnvConfig = BaseConfig & {
   frontendUrl: string;
+  serverUrl: string;
```

Comment:
Does this include the port and the protocol? If so, we have a coupling between two config fields (`port` and `serverUrl`) and should modify our code to account for it.

Reply:
- So we need the server URL to be able to implement OAuth successfully. In dev, we need the default local server URL to be set automatically e.g. `http://localhost:3001`. If the user can change the port to `6000` then it would be good if we set the `serverUrl` to `http://localhost:6000`.

- In the production use case, the port start being meaningless since they use a domain name e.g. if their port is `7000` for some reason, the server URL will still be the same e.g. `https://my-server.com`

- This means, we need to make sure that. the default dev server is set according to the port, but in the production case we should just use the user env var.